modules/global/src/de/diedavids/cuba/ceuesr/entity/Comment.java

```diff
@@ -0,0 +1,52 @@
+package de.diedavids.cuba.ceuesr.entity;
+
+import com.haulmont.chile.core.annotations.MetaProperty;
+import com.haulmont.cuba.core.entity.StandardEntity;
+import de.diedavids.cuba.entitysoftreference.EntitySoftReferenceConverter;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+@Table(name = "CEUESR_COMMENT")
+@Entity(name = "ceuesr_Comment")
+public class Comment extends StandardEntity {
+    private static final long serialVersionUID = -9012667801470279750L;
+
+    @NotNull
+    @Column(name = "TEXT", nullable = false)
+    protected String text;
+
+    @Column(name = "USERNAME")
+    protected String username;
+
+    @Basic
+    @NotNull
+    @MetaProperty(datatype = "EntitySoftReference", mandatory = true)
+    @Column(name = "COMMENTS", nullable = false)
+    @Convert(converter = EntitySoftReferenceConverter.class)
+    protected Commentable comments;
```

Comment:
The entity uses the `Commentable` interface while leveraging the *Converter and *Datatype classes from the addon.

It is also treated as `@Basic` to not screw up JPAs understanding of the relationsship (not sure why it is not relevant for Entity before, though).